clients/python/agentic-sandbox-client/sandbox-router/sandbox_router.py

```
@@ -107,14 +224,171 @@ def _env_var_is_truthy(name: str) -> bool:
+    except ValueError:
+        raise RoutingError("Invalid port format.")
+
+    pod_ip = headers.get("X-Sandbox-Pod-IP")
+    if pod_ip:
+        try:
+            ip = ipaddress.ip_address(pod_ip)
+            if ip.is_loopback or ip.is_link_local or ip.is_multicast or ip.is_unspecified:
+                raise RoutingError("Invalid target IP address.")
+            target_host = f"[{ip}]" if ip.version == 6 else str(ip)
+        except ValueError:
+            raise RoutingError("Invalid target IP address format.")
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)